### PR TITLE
Resolve Windows FS Issues

### PR DESCRIPTION
**Pull request by trevor-scheer**
This PR should resolve the codegen issues on Windows that have been reported in the long-running codegen issue thread #719

Thanks to @patrys for really digging in and suggesting some changes!

Pending set up of Appveyor, I'd appreciate if any of our Windows users could test out this change.
cc: @chenglabs @ZenSoftware @jbaxleyiii @patrys

### CODE REVIEW

File: packages/apollo/src/Command.ts [hunk 1]
```diff
@@ -188,8 +189,8 @@ export abstract class ProjectCommand extends Command {
     // In this case, we don't want to look to the .dir since that's the parent
     const rootURI =
       filepath === process.cwd()
-        ? `file://${filepath}`
-        : `file://${parse(filepath).dir}`;
+        ? Uri.file(filepath)
```
Comment (jbaxleyiii) [inline packages/apollo/src/Command.ts]: YAS

File: packages/apollo/src/commands/client/__tests__/__snapshots__/generate.test.ts.snap [hunk 2]
```diff
@@ -37,69 +37,6 @@ exports[`client:codegen generates operation IDs for swift files when flag is set
 }"
 `;
 
-exports[`client:codegen writes TypeScript global types to a custom path when globalTypesFile is set 1`] = `
```
Comment (jbaxleyiii) [inline packages/apollo/src/commands/client/__tests__/__snapshots__/generate.test.ts.snap]: Why was this removed?

Comment (trevor-scheer) [reply packages/apollo/src/commands/client/__tests__/__snapshots__/generate.test.ts.snap]: This was a result of muddying up, then deleting, and finally regenerating this test snap. The test is skipped, so the snap is obsolete. Side note: doesn't jest normally find these and remove them with `-u`?

Can totally revert this change!

**Comment by steabert** [comment]
Is there a way to apply this fix easily while waiting for the next apollo version? I'm not familiar with specifying a github url for packages belonging to a monorepo, maybe someone here knows a way to do that?

**Comment by TimKnipe** [comment]
What I've been doing is go into node_modules/apollo/lib/Command.js and find the 2 "file://" references and change them to "file:///"

Yeah, i have to redo this every time i add or update packages as it gets overwritten.

**Comment by jhalborg** [comment]
Could this please be prioritized soon? We're holding off on upgrading to 2.0+ pending this change, which again halts further work using some new Apollo Platform features.

**Comment by JakeDawkins** [comment]
@jhalborg @steabert @TimKnipe @treylon @patrys Thanks for all the feedback! These changes should be live now in version **2.2.0**!

**Comment by JakeDawkins** [comment]
@jhalborg @steabert @TimKnipe @treylon @patrys sorry, I had to unpublish because of a config issue. I'll try to get it back out today!

**Comment by NateScarlet** [comment]
Not work with `apollo/2.3.0 win32-x64 node-v10.9.0`
Still got error when mkdir
```
E:\Dev\taskboard [master ↑1 +0 ~76 -0 | +0 ~3 -0 !]> npx apollo client:codegen --globalTypesFile frontend/src/types/graphql-global.ts --target typescript types
// rootPath: E:\Dev\taskboard
// toPath(sourcePath): /e:/Dev/taskboard/frontend/src/graphql/acceptanceNode.gql
  √ Loading Apollo Project
  × Generating query files with 'typescript' target
    → ENOENT: no such file or directory, mkdir '..\..\e:\Dev\taskboard\frontend\src\graphql\types'
// rootPath: E:\Dev\taskboard
// toPath(sourcePath): /e:/Dev/taskboard/frontend/src/graphql/acceptanceNode.gql
(node:10344) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, mkdir '..\..\e:\Dev\taskboard\frontend\src\graphql\types'
```

**Comment by JakeDawkins** [comment]
@jhalborg Sorry for not responding. My notification settings are messed up 🙃

I released these changes with 2.3.0, but you may run into @NateScarlet's issue. I'll look into that and maybe cut a patch release for it if needed

**Comment by JakeDawkins** [comment]
@jhalborg 2.3.1 should have the fixes for the issue mentioned above!
